frontend/javascripts/oxalis/geometries/helper_geometries.ts

```diff
@@ -0,0 +1,164 @@
+    this.finalizeMesh();
+    app.vent.trigger("rerender");
+  }
+
+  finalizeMesh() {
+    const mesh = this.line;
+    if (mesh.geometry.attributes.position.array !== this.vertexBuffer.getBuffer()) {
+      // Need to rebuild Geometry
+      const positionAttribute = new THREE.BufferAttribute(this.vertexBuffer.getBuffer(), 3);
+      positionAttribute.setUsage(THREE.DynamicDrawUsage);
+      mesh.geometry.dispose();
+      mesh.geometry.setAttribute("position", positionAttribute);
+    }
+
+    mesh.geometry.attributes.position.needsUpdate = true;
+    mesh.geometry.setDrawRange(0, this.vertexBuffer.getLength());
+    mesh.geometry.computeBoundingSphere();
+  }
+}
+
+const rotations = {
+  [OrthoViews.PLANE_XY]: new THREE.Euler(0, 0, 0),
+  [OrthoViews.PLANE_YZ]: new THREE.Euler(Math.PI, -(1 / 2) * Math.PI, Math.PI),
+  [OrthoViews.PLANE_XZ]: new THREE.Euler((1 / 2) * Math.PI, 0, 0),
+  [OrthoViews.TDView]: null,
+};
+
+export class RectangleGeometry {
+  color: THREE.Color;
+  centerMarkerColor: THREE.Color;
+  rectangle: THREE.Mesh<THREE.PlaneGeometry, THREE.MeshBasicMaterial>;
+  centerMarker: THREE.Mesh<THREE.PlaneGeometry, THREE.MeshBasicMaterial>;
+
+  constructor() {
+    this.color = CONTOUR_COLOR_NORMAL;
+    this.centerMarkerColor = new THREE.Color(0xff00ff);
+
+    const geometry = new THREE.PlaneGeometry(1, 1);
+    const material = new THREE.MeshBasicMaterial({
+      side: THREE.DoubleSide,
+      transparent: true,
+      opacity: 0.5,
+    });
+    this.rectangle = new THREE.Mesh(geometry, material);
+
+    const centerGeometry = new THREE.PlaneGeometry(2, 2);
+    const centerMaterial = new THREE.MeshBasicMaterial({
+      color: this.centerMarkerColor,
+      side: THREE.DoubleSide,
+      transparent: true,
+      opacity: 0.9,
+    });
+    this.centerMarker = new THREE.Mesh(centerGeometry, centerMaterial);
+
+    this.reset();
+  }
+
+  reset() {
+    this.rectangle.material.color = this.color;
+    this.centerMarker.material.color = this.centerMarkerColor;
+  }
+
+  rotateToViewport() {
+    const { activeViewport } = Store.getState().viewModeData.plane;
+    const rotation = rotations[activeViewport];
+    if (!rotation) {
+      return;
+    }
+
+    this.rectangle.setRotationFromEuler(rotation);
+    this.centerMarker.setRotationFromEuler(rotation);
+  }
+
+  setColor(color: THREE.Color) {
+    this.color = color;
+    // this.color.offsetHSL(0.0, 0, -0.3);
+    // Copy this.color into this.centerMarkerColor
+    this.centerMarkerColor.copy(this.color);
+    this.centerMarkerColor.offsetHSL(0.5, 0, 0);
+
+    this.reset();
+  }
+
+  setCoordinates(startPosition: Vector3, endPosition: Vector3) {
+    const centerPosition = V3.scale(V3.add(startPosition, endPosition), 0.5);
+    const extentXYZ = V3.abs(V3.sub(endPosition, startPosition));
+    const { activeViewport } = Store.getState().viewModeData.plane;
+    const extentUVW = Dimensions.transDim(extentXYZ, activeViewport);
+    extentUVW[2] = 2;
```

Comment:
I added the following comment: `// Set the depth-component to 1 so that we don't scale the depth by 0`. I also changed the 2 to 1, because it didn't really make sense.